lease/lessor.go

```diff
@@ -20,22 +20,18 @@ import (
 	"math"
 	"sort"
 	"sync"
-	"sync/atomic"
 	"time"
 
 	"github.com/coreos/etcd/lease/leasepb"
 	"github.com/coreos/etcd/mvcc/backend"
-	"github.com/coreos/etcd/pkg/monotime"
 )
 
-const (
-	// NoLease is a special LeaseID representing the absence of a lease.
-	NoLease = LeaseID(0)
-
-	forever = monotime.Time(math.MaxInt64)
-)
+// NoLease is a special LeaseID representing the absence of a lease.
+const NoLease = LeaseID(0)
 
 var (
+	forever = time.Time{}
```

Comment:
add a comment here to explain why `forever = time.Time{}`.

Reply:
@fanminshi  How about this?
"Since time.Time with monotonic time only returned by time.Now, we can't create a max time.Time with monotonic time. Instead, use zero time.Time to mean forever and return max time.Duration when calculating its remaining."

Reply:
that's too detailed. I am thinking about just say `forever is represented by zero time.Time`. Defer to @heyitsanthony for best comment.

Reply:
Maybe use an explicit flag `isForever`, instead of a `time.Time` variable?

```
type expiry struct{
	time.Time
	isForever bool // set when lessor is demoted
}

func (l *Lease) Remaining() time.Duration {
	l.expiryMu.RLock()
	defer l.expiryMu.RUnlock()
	if l.expiry.isForever {
		return time.Duration(math.MaxInt64)
	}
	return l.expiry.Sub(time.Now())
}
```

Reply:
Please just use time.IsZero(); a separate type is excessive. The comment can be:
```go
expiry time.Time // no expiration when expiry.IsZero() is true
```

Reply:
Got it.